server/routes/authRoutes.js

```diff
@@ -0,0 +1,34 @@
+const authRouter = require('express').Router();
+const passport = require('passport');
+const requireLogin = require('../middlewares/requireLogin');
+
+authRouter.get('/facebook', passport.authenticate('facebook',
+    { scope: ['email'] }
+  )
+);
+
+authRouter.get('/facebook/callback', passport.authenticate('facebook'),
+  (req, res) => {
+    res.redirect('/viewreports');
+});
+
+authRouter.get('/google', passport.authenticate('google',
+    { scope: ['email'] }
+  )
+);
+
+authRouter.get('/google/callback', passport.authenticate('google'),
+  (req, res) => {
+    res.redirect('/viewreports');
+});
+
+authRouter.get('/current_user', requireLogin, (req, res) => {
```

Comment:
i dont get this part, what is this route doing ?

Reply:
@Karyum this route checks if the user is logged in and redirects to the 'View Reports' page if he/she is...